pennylane/devices/default_clifford.py

```diff
@@ -0,0 +1,650 @@
+            _wrap_simulate = partial(self.simulate, debugger=None)
+            with concurrent.futures.ProcessPoolExecutor(max_workers=max_workers) as executor:
+                exec_map = executor.map(
+                    _wrap_simulate,
+                    vanilla_circuits,
+                    seeds,
+                )
+                results = tuple(exec_map)
+
+            # reset _rng to mimic serial behavior
+            self._rng = np.random.default_rng(self._rng.integers(2**31 - 1))
+
+        if self.tracker.active:
+            self.tracker.update(batches=1)
+            self.tracker.record()
+            for i, c in enumerate(circuits):
+                qpu_executions, shots = get_num_shots_and_executions(c)
+                res = np.array(results[i]) if isinstance(results[i], Number) else results[i]
+                if c.shots:  # pragma: no cover
+                    self.tracker.update(
+                        simulations=1,
+                        executions=qpu_executions,
+                        results=res,
+                        shots=shots,
+                        resources=c.specs["resources"],
+                    )
+                else:
+                    self.tracker.update(
+                        simulations=1,
+                        executions=qpu_executions,
+                        results=res,
+                        resources=c.specs["resources"],
+                    )
+                self.tracker.record()
+
+        return results[0] if is_single_circuit else results
+
+    def supports_derivatives(
+        self,
+        execution_config: Optional[ExecutionConfig] = None,
+        circuit: Optional[QuantumTape] = None,
+    ) -> bool:
+        """Check whether or not derivatives are available for a given configuration and circuit.
+
+        ``DefaultClifford`` returns trivial derivates everytime.
+
+        Args:
+            execution_config (ExecutionConfig): The configuration of the desired derivative calculation
+            circuit (QuantumTape): An optional circuit to check derivatives support for.
+
+        Returns:
+            Bool: Whether or not a derivative can be calculated provided the given information
+
+        """
+
+        return True
+
+    def compute_derivatives(
+        self,
+        circuits: QuantumTape_or_Batch,
+        execution_config: ExecutionConfig = DefaultExecutionConfig,
+    ):
+        is_single_circuit = False
+        if isinstance(circuits, QuantumScript):
+            is_single_circuit = True
+            circuits = [circuits]
+
+        if self.tracker.active:
+            self.tracker.update(derivative_batches=1, derivatives=len(circuits))
+            self.tracker.record()
+
+        res = tuple(tuple(np.zeros(s) for s in circuit.shape(self)) for circuit in circuits)
+
+        return res[0] if is_single_circuit else res
+
+    # pylint: disable=unidiomatic-typecheck, unused-argument, too-many-branches, no-member, too-many-statements
```

Comment:
Anything we can do to push down the number of warnings we have to suppress?

Suppressing these warnings should be limited to either "pylint is being stupid, and what I'm doing here is actually good practice" (like unused arguments when we must match an API), or "I know what I'm doing is bad but there is no feasible way to fix this in the time that we have" (`interfaces/execution.py`).

For the below code, I think it is well within the scope of this project to fix `unidiomatic-typecheck`, `unused-argument`, `too-many-branches`, and `too-many-statements`. Breaking this function up into smaller units will help both readability and testability.

Reply:
tidied up, only one left really is `no-member` because `stim` is not imported at top-level so pylint doesn't know what its members are